### PR TITLE
feat(core): override url request option

### PR DESCRIPTION
**Pull request by jtherin**

### CODE REVIEW

File: scw/request_option.go [hunk 1]
```diff
@@ -24,6 +24,13 @@ func WithAllPages() RequestOption {
 	}
 }
 
+// WithAPIURLRequest overrides the client API URL to the given URL in the request.
+func WithAPIURLRequest(apiURL string) RequestOption {
```
Comment (jerome-quere) [inline scw/request_option.go]: Should we rename it to `WithRequestAPIURL` ? what do you think @QuentinBrosse

Comment (jtherin) [reply scw/request_option.go]: The ability to clone a client would be nice and could replace this.

Comment (QuentinBrosse) [reply scw/request_option.go]: `WithRequestAPIURL` is better yep.

**Comment by QuentinBrosse** [comment]
> Another approach to this problem would be somethibng like
> 
> ```
> client := scw.NewClient()
> client2 := client.Clone(scw.WithApiURL())
> ```
> @QuentinBrosse any thought ?

@jerome-quere @jtherin 

What is the exact use-case and why [client.WithAPIURL](https://pkg.go.dev/github.com/scaleway/scaleway-sdk-go@v1.0.0-beta.6.0.20200324130151-e0b5fd35d2c8/scw?tab=doc#WithAPIURL) is not suitable?

**Pull request closed without merging**